specs/agents/span-links-tmp.md

```diff
@@ -0,0 +1,24 @@
+TODO: this content will be appended to "span-links.md" when that file is added in https://github.com/elastic/apm/pull/600
+
+### API
+
+Agents MAY provide a public API to add span links at span/transaction creation.
+A use-case for user's manually adding span links is for [batch message processing](tracing-instrumentation-messaging.md#batch-message-processing)
+that the APM agent does not or cannot instrument. (For some agents it would be
+a burden to internally support span links and *not* expose the API publicly.)
+
+TODO: Would it be a pain to have to expose this publicly for the Java agent? Would it mean adding additional overloaded `startSpan` and `startExitSpan` methods? If so, and given an ultimate goal of preferring the OTel API, then there isn't a strong reason to *require* a public API for adding span links.
```

Comment:
We just recently discussed a related issue with the Java agent team and the conclusion was that we don't want to invest in the proprietary API anymore. So for Java, we'd rely on the OTel bridge as the only option for a public span links API. The bridge would then delegate to the internal span links API.